src/components/timeline.js

```diff
@@ -0,0 +1,385 @@
+        color: '#676767',
+        height: 9,
+    },
+    thumb: {
+        height: 34,
+        width: 34,
+        zIndex: 3,
+        backgroundColor: '#2f2f2f',
+        backgroundImage: `url(${ClockIcon})`,
+        backgroundSize: 'contain',
+        marginTop: -12,
+        marginLeft: -12,
+        '&:focus,&:hover,&$active': {
+            boxShadow: 'inherit',
+        },
+    },
+    active: {},
+    track: {
+        height: 12,
+        border: '1px solid white',
+        zIndex: 2,
+    },
+    mark: {
+        backgroundColor: 'white',
+        height: 20,
+        width: 3,
+        marginTop: -3,
+        zIndex: 3,
+    },
+    rail: {
+        height: 12,
+        border: '1px solid white',
+        opacity: 1,
+        backgroundColor: '#b5b5b5',
+        zIndex: 2,
+    },
+    markLabel: {
+        top: 40,
+        width: 35,
+        wordWrap: 'break-word',
+        whiteSpace: 'normal',
+        lineHeight: 1,
+        textAlign: 'center',
+        marginLeft: 4,
+    },
+    valueLabel: {
+        left: 0,
+    },
+    [theme.breakpoints.down('xs')]: {
+        root: {
+            marginTop: 30,
+        },
+        mark: {
+            display: 'none',
+        },
+        track: {
+            height: 9,
+            borderTopWidth: 0,
+        },
+    },
+}))(Slider);
+
+let mergeListByHours = [];
+
+const Timeline = (props) => {
+    const merges = useSelector((state) => state.merges);
+
+    const websocketExpectedCloseRef = useRef();
+
+    const [showMessage, setShowMessage] = useState(false);
+
+    const dispatch = useDispatch();
+
+    const classes = useStyles();
+
+    const minHour = 'T00:00:00Z';
+
+    const maxHour = 'T23:59:59Z';
+
+    function updateIgm(tso, status) {
+        dispatch(updateIgmStatus(props.name, tso.toLowerCase(), status));
+    }
+
+    function toIgmStatus(status) {
+        switch (status) {
+            case 'AVAILABLE':
+                return IgmStatus.AVAILABLE;
+
+            case 'VALIDATION_SUCCEED':
+                return IgmStatus.VALID;
+
+            case 'VALIDATION_FAILED':
+                return IgmStatus.INVALID;
+
+            case 'BALANCE_ADJUSTMENT_SUCCEED':
+            case 'LOADFLOW_SUCCEED':
+                return IgmStatus.MERGED;
+
+            case 'BALANCE_ADJUSTMENT_FAILED':
+            case 'LOADFLOW_FAILED':
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Return current date in format YYYY-MM-DD, example: 2020-09-23
+     * @returns {string}
+     */
+    const currentDateFormat = () => {
+        const date = new Date();
+        return date.toISOString().substr(0, 19).split('T')[0];
+    };
+
+    /**
+     * Return array with hours convert to minutes
+     */
+    const mergesForTimeline = (merges) => {
+        mergeListByHours = [];
+        if (merges) {
+            setShowMessage(false);
+            merges.forEach((merge) => {
+                const hour = merge.date.split('T')[1].split(':')[0];
```

Comment:
I guess there is a more elegant way to get hour and day. Is there something like this in js Date? In another external lib?

Reply:
ok i will test this